core/src/main/java/org/apache/james/core/healthcheck/HealthCheck.java

```diff
@@ -18,8 +18,18 @@
  ****************************************************************/
 package org.apache.james.core.healthcheck;
 
+import org.reactivestreams.Publisher;
+
+import reactor.core.publisher.Mono;
+
 public interface HealthCheck {
     ComponentName componentName();
 
-    Result check();
+    default Result check() {
+        return Mono.from(checkReactive()).block();
+    }
+
+    default Publisher<Result> checkReactive() {
```

Comment:
if I create an implementation of HealthCheck without override check or checkReactive, I have an infinite loop?

Reply:
Yes.

Reply:
Yes, i am aware of this. I'll fix it by making all the healthChecks reactive and keep only 1 method. May be on next PR .

Reply:
fixed with: https://github.com/linagora/james-project/pull/3274